python_modules/dagster-graphql/dagster_graphql/implementation/pipeline_run_storage.py

```diff
@@ -1,27 +1,102 @@
+import os
+from enum import Enum
+from threading import Event, Thread
+from time import sleep
+
+import gevent
+from dagster import check
+
+
+class State(Enum):
+    # initialized
+    NULL = 0
+    # streaming the existing event log entries
+    LOADING = 1
+    # watching for new writes
+    WATCHING = 2
+
+
+def get_chunk_size() -> int:
+    return int(os.getenv("DAGIT_EVENT_LOAD_CHUNK_SIZE", "10000"))
+
+
 class PipelineRunObservableSubscribe:
     def __init__(self, instance, run_id, after_cursor=None):
         self.instance = instance
         self.run_id = run_id
         self.observer = None
+        self.state = State.NULL
+        self.load_thread = None
+        self.stopping = None
         self.after_cursor = after_cursor if after_cursor is not None else -1
 
     def __call__(self, observer):
         self.observer = observer
-
-        events = self.instance.logs_after(self.run_id, self.after_cursor)
+        check.invariant(self.state is State.NULL, f"unexpected state {self.state}")
+        chunk_size = get_chunk_size()
+        events = self.instance.logs_after(self.run_id, self.after_cursor, limit=chunk_size)
         if events:
             self.observer.on_next(events)
```

Comment:
opted to maintain the existing load on init pattern to reduce who is effected by this change to only those with large event logs